src/main/java/uk/gov/hmcts/reform/idam/service/StaleUsersService.java

```diff
@@ -25,13 +26,20 @@ public class StaleUsersService {
     private final IdamClient client;
     private final ParameterResolver parameterResolver;
 
+    @Retry(retryAttempts = 2)
     public List<String> fetchStaleUsers() {
-        StaleUsersResponse staleUsersResponse = client.getStaleUsers(
-            Map.of(
-                PAGE_NUMBER_PARAM, currentPage,
-                BATCH_SIZE_PARAM, parameterResolver.getBatchSize()
-            )
-        );
+        final StaleUsersResponse staleUsersResponse;
+        try {
+            staleUsersResponse = client.getStaleUsers(
+                Map.of(
+                    PAGE_NUMBER_PARAM, currentPage,
+                    BATCH_SIZE_PARAM, parameterResolver.getBatchSize()
+                )
+            );
+        } catch (Exception e) {
+            log.error("StaleUsersService.getStaleUsers threw exception: {}", e.getMessage(), e);
+            throw e;
```

Comment:
Can you please rethrow a custom exception instead, StaleUsersServiceException for example.